app/forms/schools/add_participant_form.rb

```diff
@@ -79,7 +79,12 @@ def mentor
     end
 
     def email_already_taken?
-      User.find_by(email: email)&.participant_profiles&.ecf&.any?
+      User.find_by(email: email)
+        &.teacher_profile
+        &.participant_profiles
```

Comment:
do we want `participant_profiles through teacher_profile` on user model? perhaps not, up for discussion

Reply:
Think that's a topic for 2pm 🙂

Reply:
Eeeeh, I have a clash 😢 

In general I am not sure how true the code comment in User model
```
# TODO: Legacy associations, to be removed
```
is. There are some parts of our app that want profiles and users to be closely related - in particular Track and Pay apis talk about 'participants' - which are users - and there is quite a bit of logic around validating them that wants to know which profiles that user has. Maybe at some point we will switch them to operate on teacher profiles instead, but, ehh, not sure when that would be?

Reply:
yesterday 2pm never ended up happening. I still think this is a wider conversation than we can have here